File: NmsCinema/src/app/movie.ts
import { Category } from "./category";

export class Movie {

    name : string;
    id:string;
    duration:string;
    quantity:number;
    qty:number;
    price:number;
    activate:boolean;
    description:string;
    category:Category;
    starCast:string;

    
    constructor(name:string,duration:string,quantity:number,price:number,activate:boolean,description:string,starCast:string,category:Category)
    {


        this.name  = name;
        this.duration = duration;
        this.quantity = quantity;
        this.price = price;
        this.activate = activate;
        this.description = description;
        this.category = category;
        this.starCast = starCast;
    }
}

File: CinemaBackEnd/medicine/src/main/java/com/medicare/medicine/model/Movie.java
package com.medicare.medicine.model;


import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
@Table
public class Movie {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Integer id;

    private String name;
    private String description;
    private String duration;
    private Integer quantity;
    private Integer qty;
    private Double price;
    @Column(columnDefinition =  "boolean default true ")
    private boolean activate;
    private String cast;
    @OneToOne
    @JoinColumn(columnDefinition = "category_Id")
    private Category category;

}
